File: packages/scraper/app/models/execution_log.py
from sqlalchemy import Column, String, Text, Boolean, Integer, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID, JSONB
from app.database import Base
import uuid


class ExecutionLog(Base):
    __tablename__ = 'execution_logs'

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    task_id = Column(UUID(as_uuid=True), ForeignKey('tasks.id'), nullable=False)
    started_at = Column(DateTime, nullable=True)
    completed_at = Column(DateTime, nullable=True)
    status = Column(String(20), default='pending')
    is_dry_run = Column(Boolean, default=False)
    retry_count = Column(Integer, default=0)
    error_message = Column(Text, nullable=True)
    screenshot_path = Column(Text, nullable=True)
    steps_log = Column(JSONB, nullable=True)
    vnc_session_id = Column(String(100), nullable=True)
    created_at = Column(DateTime)

File: packages/backend/app/Models/ExecutionLog.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ExecutionLog extends Model
{
    use HasFactory, HasUuids;

    /**
     * The primary key type.
     */
    protected $keyType = 'string';

    /**
     * Indicates if the IDs are auto-incrementing.
     */
    public $incrementing = false;

    /**
     * The attributes that are mass assignable.
     */
    protected $fillable = [
        'task_id',
        'started_at',
        'completed_at',
        'status',
        'is_dry_run',
        'retry_count',
        'error_message',
        'screenshot_path',
        'steps_log',
        'vnc_session_id',
    ];

    /**
     * Get the attributes that should be cast.
     */
    protected function casts(): array
    {
        return [
            'is_dry_run' => 'boolean',
            'steps_log' => 'array',
            'started_at' => 'datetime',
            'completed_at' => 'datetime',
        ];
    }

    /**
     * Get the task that owns the execution log.
     */
    public function task(): BelongsTo
    {
        return $this->belongsTo(Task::class);
    }
}
